stream/stream.go

```diff
@@ -4,36 +4,36 @@ package stream
 // In DDD terms, it is the "aggregate root".
 //
 // In order for your domain type to implement Stream:
-//   * Embed Events
-//   * Implement `StreamID` returning a unique identifier (usually the same as your stream's internal ID).
-//   * Implement `PopEvents` that returns the events on the Events.
-//   * Implement `FromEvents` to apply events to your stream.
+//   - Embed pointer to Events queue.
+//   - Implement `StreamID` returning a unique identifier (usually the same as your stream's internal ID).
+//   - Implement `Events` to expose the Events queue.
```

Comment:
This comes with the downside we discussed a while ago: `Events()` and all its events become public, so it's possible to `Record` an event outside of the domain model.

Reply:
Not sure how big an issue this is. @roblaszczak @maclav3?

Reply:
🤔 Yes, that is right. Maybe we could use a simplified interface here. I think I have an idea 💡.

Reply:
nice one! In theory, you still can call `stream.Record` outside of the domain model, but I don't think you would do that by accident.

Reply:
On the other hand, It is a bit odd that you are using some type of property in your own struct and you have to use some library function in order to interact with that 🤔 . Maybe having `.Register(...)` method public would be better and we can just accept the risk that comes with it?

Reply:
Option with public method: https://github.com/ThreeDotsLabs/esja/pull/14/commits/e62efa70fb553853c4aebfb98bf7dae67e9af0df

Reply:
You would need to call `stream.Record` anyway, as it calls `ApplyTo`. Or call `ApplyTo` yourself.

But yeah, perhaps the public method is fine. `stream.Record` in this case is just a helper but you don't have to use it if you don't want to. And if you call `Events().Record()` outside of the domain model, it's on you